### DISCUSSION TITLE
Improvement in the development of actions in Snackbars

### DISCUSSION THREAD

**Discussion by Leycarr** [Telefonica/mistica-design#1681] [Proposals]
### Goal

In the designs we have to implemented the snackbars in an adhoc way, because the actions that the component itself provides only allow closing or dismissing the information. Just comment that if the improvement of being able to perform an action was implemented in the development, the component would improve.

### Context

 In the figma component, you can put two types of actions, long and short, but none of the development appears according to what the team tells me about the possibility of actually adding an action other than closing

### Impact

This would make future updates easier for developers.

### Discussion Questions

Could this improvement be made?

**Comment by aweell** [comment]
Hi @Leycarr,

Thank you for opening a discussion.

From the specs:

> * **Action Button:** Always contains text. Optional. This button, in addition to performing an action, always closes the Snackbar.
> * **Dismiss:** Cases:
>   * Duration: Infinite | No Action Button: In cases where the Snackbar duration is infinite and there is no action button, a dismiss option will always be shown to allow users to close the Snackbar.
>   * Duration: Infinite | With Action Button: If there is an action button and the duration is set to infinite, a dismiss option may optionally be displayed.
>   * Duration: 5s / 10s | With/Without Action Button: In cases where the duration is different from infinite, whether or not there is an action button, a dismiss option may optionally be shown. 

So the ways the snackbar action can be used:

* To dismiss the snackbar
* To perform an action and close the snackbar

What exactly are you trying to accomplish here?

[Here's an example with action + dismiss](https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhgNoBOEA7gDICW1ANBhTFQErN21ALrYM-GAEMwVAHQBXXgGUqUqjGS1ipUsAxU2VADYx4GAiACSGgLYYAjJe5ty5gDxXyAUUiSA5grGUvQYAPR4AL6cOrr6hiZmFtZ2GABMzhiuEEQeXjkAwiEBQSHhUTEkcQZGpuaWNjD2AMyZ2bkYnuQFxhAKUCXBoREY0bEiaADcxLHk1HT0TRAAbjBQAAqMUAqy3LwCS6sbWztUYlgS0rKKKmoayERBxlOzOfO0CkRQEHt8AKpfCDnS4yeRKGCqdSaABmUmMvBeRFelBoAAspF9TAARGCmDSCJjiZBsL4wAAemFw%2BFipDmNAUAAcoFCBswKOJaHIuYxWBwztMqrS3jRFrYVmtGvYLozmRpWUwKHIKAzjK5NCTYGTuA40LQAAwiAW6Hh8AnCKjIGUsgkURHG-aSMVHTbQU7IUXiqCSu2jAXI%2BbozEwAHfImUvCEQVZaEYd2HNYu7ayTCR42kB1CPmoBmLZY28N0LlyHMwPNs7ge50nWQTI1ph3x46u2QPJ4%2B%2B3-QHIACEn2%2B7dGsUifqqiyoCnoJGQNM6OBnpHcZr5c6jxuAPPNclsUgZqB5LlJFOweGnq7Ti%2BY87TpAA1jAAJ5YYAa8mRK-XqQUfZPnlydpvs9rwSUwf2YORgJgADr2veg2H8VEqCfd9oPcKA2GWZDoJ4Kh7xA4BUywwjURgOCEPqEAHD1PUAFJnEwwi0OVYJ73I6FTDJOjAMI0g4TgohJQociwBgIgNHoTjuOvSIoMk0gV1ktcwgAKgwP5eAwKQMGhT5ZDYHIDAgMgd3HRYDGIsgJ0WUSMGWOEFBgDAIBjF8yQwJSwhkhTOnyIgACEFCoKh9IwHynx8gAVehP1RIoiEGEJIgMe8GRgLBLGZIh-BgcSQHorCck2GAvyfdBj2pLjZMDKBsVxPgYAJYlDwHLz007b4ez7CBmoU6S8rTeToL6413DCHz-MCnIBtkka0OWKbCM86CImQtA0EWhcwiXah5r0TqMAAMn22NkPcZQiBkG8ACMQno2wiooKQsrSlImgwSs1ksejLoCoKiHC8lEMsEMIE%2BiqMCYIxUSxDhbA4Ch6MKXpeBK-RQT0khIgLAjuLYGNkDR-SsCJ5JfL%2BcLwoAeQAOUsFMhowKrTGB9A6249boNx2MCZIImLksLErGUABZQXlFp8qvP2Zne0BbqpPo3quJwEazou676B2jA1pnEaBsRTH0AAkBOBAKhiLuigEFoEAADV0IgAB9KmYCYEARBNiGoDNy34FoZoADYDUiIA)
